.github/workflows/workflow.yml

```diff
@@ -1,25 +1,70 @@
 name: Main workflow
-on: [push]
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - releases/*
+
 jobs:
-  run:
-    name: Run
+  build:
+    name: Build
     runs-on: ${{ matrix.operating-system }}
     strategy:
       matrix:
         operating-system: [ubuntu-latest, windows-latest]
     steps:
-    - uses: actions/checkout@master
-    
-    - name: Set Node.js 10.x
-      uses: actions/setup-node@master
-      with:
-        version: 10.x
+      - uses: actions/checkout@v2
+      
+      - name: Setup node 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+
+      - name: npm install
+        run: npm install
+
+      - name: Lint
+        run: npm run format-check
+
+      - name: npm test
+        run: npm test
+
+  test:
+    name: Test E2E
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup node 10
+        uses: ./
+        with:
+          node-version: 10.x
 
-    - name: npm install
-      run: npm install
+      - name: Verify node version
+        run: __tests__/verify-node-version.sh 10
+
+  test-proxy:
+    name: Test E2E with proxy
+    runs-on: ubuntu-latest
+    container:
+      image: alpine:latest
+      options: --dns 127.0.0.1
+    services:
+      squid-proxy:
+        image: datadog/squid:latest
+        ports:
+          - 3128:3128
+    env:
+      https_proxy: http://squid-proxy:3128
+    steps:
+      - uses: actions/checkout@users/ericsciple/m165proxy
```

Comment:
switch to `@v2` after it's published

Reply:
... and by that you mean checkout v2.  yep.

Reply:
yes